File: 0445.AddTwoNumbersII.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    s1, s2 := toNum(l1), toNum(l2)
    c := 0
    var head *ListNode
    for len(s1) > 0 || len(s2) > 0 || c > 0 {
        if len(s1) > 0 {
            c += s1[len(s1)-1]
            s1 = s1[:len(s1)-1]
        }
        if len(s2) > 0 {
            c += s2[len(s2)-1]
            s2 = s2[:len(s2)-1]
        }
        node := &ListNode{Val: c%10, Next: head}
        head = node
        c /= 10
    }
    return head
}

func toNum(l *ListNode) []int {
    stack := []int{}
    for l != nil {
        stack = append(stack, l.Val) 
        l = l.Next
    }
    return stack
}

File: 0000-0999/0445.AddTwoNumbersII.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        s1, s2 = self.toNum(l1), self.toNum(l2)
        c = 0
        head = None
        while s1 or s2 or c:
            if s1:
                c += s1.pop()
            if s2:
                c += s2.pop()
            tmp = ListNode(c%10)
            tmp.next = head
            head = tmp
            c /= 10
        return head
        
    def toNum(self, l):
        stack = []
        while l:
            stack.append(l.val)
            l = l.next
        return stack